### PR TITLE
Add support for L4S

### PR DESCRIPTION
**Pull request by hawkinsw**

### CODE REVIEW

File: cmd/networkqualityd/main.go [hunk 1]
```diff
@@ -53,6 +53,9 @@ var (
 
 	socketSendBuffer = flag.Uint("socket-send-buffer-size", 0, "The size of the socket send buffer via TCP_NOTSENT_LOWAT. Zero/unset means to leave unset")
 
+	withL4s          = flag.Bool("with-l4s", false, "Enable L4S using the default congestion control algorithm")
```
Comment (randall) [inline cmd/networkqualityd/main.go]: Can you rename these to enable-...? That's the terminology I've gone with in the other options.

And instead of an empty withL4sAlgorithm, default it to "Prague" in the flag set. That'll allow for a more explicit default in the help (and cleanup some of the setup below).

**Comment by randall** [comment]
Thank you for adding this!

File: cmd/networkqualityd/main.go (hunk 1 above)
Comment (hawkinsw) [reply cmd/networkqualityd/main.go]: Great!

Comment (hawkinsw) [reply cmd/networkqualityd/main.go]: Unfortunately, the built-in `flag` package in go leaves something to be desired. The semantics of a default value for a string variable do not work as you would expect. Here's how I would expect it to work:

If I set a flag named `enable-l4s` and gave it the `prague` default value, I would expect that
```
$ ./app ... -enable-l4s ....
```

would be a-okay -- the flag variable would just contain the default value. Similarly, I would expect that
```
$ ./app ... -enable-l4s bbrv2 ...
```

would be a-okay, too -- the flag variable would contain `bbrv2` rather than the default value.

I would then use some of the other tools to determine whether the user actually passed that flag in order to differentiate between the flag having the default value because the user didn't specify the option or because they specified the option and wanted the default value (that workaround is no big deal).

*However*, that's not how it works:

```console
$ ./app ... -enable-ls ...
````

is not allowed. 

So, long story short, I couldn't clean up the code the way you recommended or the way that I wanted. Sorry!

**Review by randall — APPROVED** [review]
